### PR TITLE
Fix the crash while calling the payment handlers.

### PR DESCRIPTION
**Pull request by prasanthu**
Read all the comments before implementing new design. The way I implemented will make it crash ( call the the methods `paymentAuthorizationViewController` recursively )

### CODE REVIEW

File: ostelco-ios-client/Protocols/ApplePayDelegate.swift [hunk 1]
```diff
@@ -38,6 +38,7 @@ extension ApplePayError: LocalizedError {
     }
 }
 
+// swiftlint:disable:next class_delegate_protocol
```
Comment (designatednerd) [inline ostelco-ios-client/Protocols/ApplePayDelegate.swift]: Blergh, that's probably a swift 5 bug on swiftLint - now that you can specify that anything can be a protocol constraint (as UIViewController is here), SwiftLint keeps looking for the `ApplePayDelegate: class` declaration even though it's no longer necessary.

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift [hunk 2]
```diff
@@ -24,14 +24,14 @@ class ApplePayViewController: UIViewController {
     var applePayError: ApplePayError?
 
 }
-extension ApplePayViewController: ApplePayDelegate {
+extension ApplePayViewController: ApplePayDelegate, PKPaymentAuthorizationViewControllerDelegate {
```
Comment (designatednerd) [inline ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: fwiw i'd break this out into separate extensions - that way you can more clearly tell which are the apple methods you're overriding

Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: Since I am only using default implementation for

Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: extension ApplePayViewController: ApplePayDelegate is empty,

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift [hunk 3]
```diff
@@ -16,15 +16,16 @@ import PassKit
 // TL;DR: @objc functions may not currently be in protocol extensions.
 // You could create a base class instead, though that's not an ideal solution.
 // https://stackoverflow.com/questions/39487168/non-objc-method-does-not-satisfy-optional-requirement-of-objc-protocol
-class ApplePayViewController: UIViewController {
+class ApplePayViewController: UIViewController, ApplePayDelegate {
```
Comment (prasanthu) [inline ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: Is this a better way ? @designatednerd

**Pull request merged**